actions/global_actions.jsx

```diff
@@ -292,8 +292,14 @@ async function getTeamRedirectChannelIfIsAccesible(user, team) {
         teamChannels = getChannelsNameMapInTeam(state, team.id);
     }
 
-    let channelName = LocalStorageStore.getPreviousChannelName(user.id, team.id);
+    const channelName = LocalStorageStore.getPreviousChannelName(user.id, team.id);
     channel = teamChannels[channelName];
+
+    if (typeof channel === 'undefined') {
```

Comment:
What happens if you're removed from a channel you're viewing? Usually we're redirected to Town Square when that happens, but from this logic it looks like it will redirect you to the last DM/GM you were in.

Reply:
This works correctly for me after my latest change.

Reply:
So upon leaving a direct channel, this function will return you to the default channel because the previous channel has been removed from the system, that makes sense.